src/hdmf/container.py

```diff
@@ -632,7 +632,7 @@ def __repr__(self):
             # if isinstance(v, DataIO) or not hasattr(v, '__len__') or len(v) > 0:
             if hasattr(v, '__len__'):
                 if isinstance(v, (np.ndarray, list, tuple)):
-                    if len(v) > 0:
+                    if (hasattr(v, 'shape') and len(np.shape(v)) == 0) or len(v) > 0:
```

Comment:
What does this whole block from 633-638 do? The `template += ...` is the same. It looks odd.

It looks like this if statement adds to the template if `v` has more than one element or is a scalar compound. But what if `v` is not a compound dtype and has only one element? Does nothing happen?

Reply:
My understanding of the original code was that it will add to the template as long as it's not an empty list/array/tuple. However for a scalar with a compound data type, `hasattr(v, '__len__')` is True but `len(v)` will error with `TypeError: len() of unsized object`.

I think if `v` is not a compound type and has at least one element it will be added to the template.... I can add a test for that to make sure?

Reply:
Even if the value is a list/array/tuple that is empty, I think we should still add it to the template. Maybe that will clutter the display, but I think it is still useful to see. What do you think? That's a bit tangential to this PR but I think it's a good improvement. It would remove the modified if statement all together.

Reply:
I think it might clutter things up... removing the if statement will make an nwbfile look like this:

![Screenshot 2024-08-21 at 3 39 58 PM](https://github.com/user-attachments/assets/c5a4e841-37c1-4034-b693-d102af481127)

vs. with the if statement:
![Screenshot 2024-08-21 at 3 40 40 PM](https://github.com/user-attachments/assets/ebd6f4fb-571f-410b-ac25-675929b53110)

Reply:
Actually disregard the previous comment, I had condensed the if statements too much. I agree if it's an empty list/array/tuple it could be useful to display.

Simplifying to this is what you're suggesting then?
```python
if isinstance(v, (np.ndarray, list, tuple)):
    template += "  {}: {}\n".format(k, self.__smart_str(v, 1))
elif v:
    template += "  {}: {}\n".format(k, self.__smart_str(v, 1))
```

Reply:
Yes, exactly. Great! Let's do that.